Mage.Sets/src/mage/cards/w/WonderscapeSage.java

```diff
@@ -0,0 +1,165 @@
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.util.CardUtil;
+
+/**
+ * @author grimreap124
+ */
+public final class WonderscapeSage extends CardImpl {
+
+    private static final FilterControlledPermanent filter = new FilterControlledLandPermanent();
+    private static final FilterControlledPermanent basicLandFilter = new FilterControlledLandPermanent();
+
+    static {
+        basicLandFilter.add(SuperType.BASIC.getPredicate());
+    }
+
+    public WonderscapeSage(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}");
+        
+        this.subtype.add(SubType.MOONFOLK);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // {T}, Return a land you control to its owner's hand: Draw a card. Then discard a card unless that land had a nonbasic land type.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new WonderscapeSageEffect(), new TapSourceCost());
+        ability.addCost(new WonderscapeSageCost(new TargetControlledPermanent(filter)));
+        this.addAbility(ability);
+    }
+
+    private WonderscapeSage(final WonderscapeSage card) {
+        super(card);
+    }
+
+    @Override
+    public WonderscapeSage copy() {
+        return new WonderscapeSage(this);
+    }
+}
+
+class WonderscapeSageEffect extends OneShotEffect {
+
+    WonderscapeSageEffect() {
+        super(Outcome.Benefit);
+        staticText = "Draw a card. Then discard a card unless that land had a nonbasic land type.";
+    }
+
+    private WonderscapeSageEffect(final WonderscapeSageEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WonderscapeSageEffect copy() {
+        return new WonderscapeSageEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        new DrawCardSourceControllerEffect(1).apply(game, source);
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        for (Cost cost :source.getCosts()) {
+            if (cost instanceof WonderscapeSageCost) {
+                if (((WonderscapeSageCost) cost).nonbasic) {
+                    return true;
+                }
+                player.discardOne(false, false, source, game);
+                return true;
+            }
+        }
+        return true;
+    }
+}
+
+class WonderscapeSageCost extends CostImpl {
```

Comment:
No need in custom cost. Update original cost instead (see attached screen with such cost) to save permanent, see example from SacrificeTargetCost:
![IMG_0545](https://github.com/magefree/mage/assets/8344157/caa397b0-c110-470d-9c0f-c4caca54716f)

Then search payed permanent from costs list:
![IMG_0546](https://github.com/magefree/mage/assets/8344157/8eb9dccf-d418-4878-b624-e0d839803671)

Reply:
Awesome I can do that tomorrow

Reply:
Original problem with Omo, Queen of Vesuva is out of scope of current PR, see #12506 and related discussion. Just ignore that problem here.